### PR TITLE
Unity 5.0 staging stv

### PR DESCRIPTION
**Pull request by UnityAlex**

### CODE REVIEW

File: external/buildscripts/build_runtime_stv.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ for STV_TARGET in ${STV_TARGETS}; do
 	STV_GCC_PREFIX="${!STV_GCC_PREFIX}"
 
 	# need to swap flags when building for different platforms.
-	CXXFLAGS="-g -DARM_FPU_VFP=1 -DHAVE_ARMV6=1 -D__ARM_EABI__ -DLINUX -D__linux__ -DHAVE_PTHREAD_MUTEX_TIMEDLOCK -march=armv7-a -mfpu=vfp -mfloat-abi=softfp -fpic -funwind-tables -ffunction-sections -fdata-sections";
+	CXXFLAGS="-g -DARM_FPU_VFP=1 -DHAVE_ARMV6=1 -D__ARM_EABI__ -DLINUX -D__linux__ -DPLATFORM_STV -DHAVE_PTHREAD_MUTEX_TIMEDLOCK -march=armv7-a -mfpu=vfp -mfloat-abi=softfp -fpic -funwind-tables -ffunction-sections -fdata-sections";
```
Comment (joncham) [inline external/buildscripts/build_runtime_stv.sh]: I think you could also just define USE_PTHREAD_LOCKS and get the same result without changing the header file. Anyone prefer one or the other?

File: mono/metadata/object.c [hunk 2]
```diff
@@ -1132,13 +1132,14 @@ add_imt_builder_entry (MonoImtBuilderEntry **imt_builder, MonoMethod *method, gu
...
 	if (e != NULL) {
+		MonoMethod *method = e->key;
 		printf ("  * %s [%d]: (%p) '%s.%s.%s'\n",
 				message,
 				num,
-				e->method,
-				e->method->klass->name_space,
-				e->method->klass->name,
-				e->method->name);
+				method,
```
Comment (joncham) [inline mono/metadata/object.c]: Unrelated cleanup change I assume?

**Comment by joncham** [comment]
Looks okay to me.

File: mono/metadata/object.c (hunk 2 above)
Comment (UnityAlex) [inline mono/metadata/object.c]: Yeah unrelated. When I was attempting to debug the IMT stuff this function did not work so I fixed it. Just pushing the fix back so other people don't have to. :-)

**Comment by UnityAlex** [comment]
Jon's solution of defining USE_PTHREAD_LOCKS works as well and is a better solution. I've switched the PR to use that instead.

**Comment by Tak** [comment]
Merged.
